plugins/dynamic-few-shot-lambda/template.yml

```diff
@@ -0,0 +1,216 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Deploy demo Lambda function for GenAI IDP dynamic few-shot prompting
+
+Parameters:
+
+  PermissionsBoundaryArn:
+    Type: String
+    Default: ""
+    Description: >-
+      (Optional) ARN of an existing IAM Permissions Boundary policy to attach to the Lambda execution role.
+      Leave blank if no Permissions Boundary is required.
+    AllowedPattern: "^(|arn:aws[a-z-]*::iam::[0-9]{12}:policy/.+)$"
+    ConstraintDescription: Must be empty or a valid IAM policy ARN
+
+  VectorBucketName:
```

Comment:
It's unclear to me where/how this VectorBucket gets created..  Is it presumed to pre-exist before deploying the plugin?  Where are the user instructions for how to create it? 
Shouldn't your stack create it, if it's not pre-created?  (Eg see how our main stack handles EvaluationBaselineBucket or ReportingBucket - user can optionally provide a precreated bucket, but if not, stack creates one automatically)